### PR TITLE
Performance bugs

### PR DESCRIPTION
**Pull request by benghaziboy**

### CODE REVIEW

File: bulbs/contributions/views.py [hunk 1]
```diff
@@ -143,9 +174,9 @@ def list(self, request):
         if format == 'csv':
             queryset = self.get_queryset()
             queryset = queryset[:queryset.count()]
-            data = {
-                'results': queryset
-            }
+            # data = {
+            #     'results': queryset
+            # }
```
Comment (mparent61) [inline bulbs/contributions/views.py]: I would either document why commented out or remove

**Comment by mparent61** [comment]
Would you summarize the improvements?

**Comment by benghaziboy** [comment]
this separates nested rates into their own endpoints to fix the save request on the frontend

File: .cache/v/cache/lastfailed [hunk 2]
```diff
@@ -0,0 +1,128 @@
+{
```
Comment (mparent61) [inline .cache/v/cache/lastfailed]: what is this file?

File: bulbs/contributions/views.py [hunk 3]
```diff
@@ -40,6 +42,35 @@ def get_queryset(self):
         return qs
 
 
+class NestedRateViewSet(viewsets.ModelViewSet):
+    """filters queryset by role given the role_pk url kwarg"""
+
+    def get_queryset(self):
+        role_pk = self.request.parser_context["kwargs"].get("role_pk")
```
Comment (MichaelButkovic) [inline bulbs/contributions/views.py]: will this 500 if `role_pk` isn't included?

**Comment by mparent61** [comment]
:+1:

File: bulbs/contributions/views.py (hunk 3 above)
Comment (benghaziboy) [inline bulbs/contributions/views.py]: role_pk is inherently in the path. View wont be accessed without it.

Comment (MichaelButkovic) [inline bulbs/contributions/views.py]: ah, i see

**Comment by MichaelButkovic** [comment]
LGTM

**Pull request merged**